File: lib/solanum/output/print.rb
class Solanum
class Output
class Print
  def initialize(args)
  end

  def write_events(events)
    events.each do |event|
      puts "%-40s %5s (%s) %s" % [
        event['service'], event['metric'],
        event['state'].nil? ? "--" : event['state'],
        event.inspect
      ]
    end
  end
end
end
end

File: src/solanum/output/print.clj
(ns solanum.output.print
  "Prints events to standard out."
  (:require
    [solanum.output.core :as output]))


(defn- print-event
  "Print the event to `*out*`."
  [event]
  (printf "%-40s %5s (%s) %s\n"
          (:service event)
          (:metric event)
          (:state event "--")
          (pr-str event)))


(defrecord PrintOutput
  []

  output/Output

  (write-events
    [this events]
    (run! print-event events)
    (flush)))


(defmethod output/initialize :print
  [config]
  (map->PrintOutput (select-keys config [:type])))
